src/Kiota.Builder/Writers/CSharp/CodeMethodWriter.cs

```diff
@@ -209,8 +209,11 @@ private void WriteConstructorBody(CodeClass parentClass, CodeMethod currentMetho
                                         .OrderByDescending(static x => x.Kind)
                                         .ThenBy(static x => x.Name)) {
             var defaultValue = propWithDefault.DefaultValue;
-            if(propWithDefault.Type is CodeType propertyType && propertyType.TypeDefinition is CodeEnum enumDefinition) {
-                defaultValue = $"{enumDefinition.Name.ToFirstCharacterUpperCase()}.{defaultValue.Trim('"').ToFirstCharacterUpperCase()}";
+            if(propWithDefault.Type is CodeType propertyType) {
```

Comment:
you should be able to add another where with something like that

```CSharp
.Where(static x => x.Type it not CodeType xType || xType.TypeDefinition is not CodeClass xClass || xClass.OriginalComposedType is null)
```
Reducing the number of loops, allowing you to merge the two ifs, reducing allocs etc...